api/apps/geoprocessing/src/modules/scenario-planning-units-inclusion/scenario-planning-units-inclusion-processor.ts

```diff
@@ -59,13 +84,50 @@ export class ScenarioPlanningUnitsInclusionProcessor
       );
     }
 
-    // After setting inclusions and exclusions from geometries above, add
-    // inclusions and exclusions by id, so that we effectively use the union
-    // of the two methods (byId and byGeoJson).
-    puIdsToInclude.push(...(job.data.include?.pu ?? []));
-    puIdsToExclude.push(...(job.data.exclude?.pu ?? []));
-    const uniquePuIdsToInclude = [... new Set(puIdsToInclude)];
-    const uniquePuIdsToExclude = [... new Set(puIdsToExclude)];
+    const puIdsToIncludeFromIds = new Set(job.data.include?.pu ?? []);
+    const puIdsToExcludeFromIds = new Set(job.data.exclude?.pu ?? []);
+
+    // If there are overlaps between opposing claims byId and byGeoJSON, ignore the claims byId
+    const puIdsToIncludeFromIdsLessIdsToExcludeFromGeo = [
+      ...new Set(
+        [...puIdsToIncludeFromIds].filter(
+          (i) => !new Set(puIdsToExcludeFromGeo).has(i),
+        ),
+      ),
+    ];
+    const puIdsToExcludeFromIdsLessIdsToIncludeFromGeo = [
+      ...new Set(
+        [...puIdsToExcludeFromIds].filter(
+          (i) => !new Set(puIdsToIncludeFromGeo).has(i),
+        ),
+      ),
+    ];
+
+    // Union of claims byId and byGeoJSON, for inclusion and for exclusions
+    puIdsToInclude.push(
+      ...[
+        ...puIdsToIncludeFromIdsLessIdsToExcludeFromGeo,
+        ...puIdsToIncludeFromGeo,
+      ],
+    );
+    puIdsToExclude.push(
+      ...[
+        ...puIdsToExcludeFromIdsLessIdsToIncludeFromGeo,
+        ...puIdsToExcludeFromGeo,
+      ],
+    );
+    const uniquePuIdsToInclude = new Set(puIdsToInclude);
+    const uniquePuIdsToExclude = new Set(puIdsToExclude);
```

Comment:
it looks like there is a high computational complexity here, especially with creating a set in every iteration of the filter.

I'd suggest to try `difference` method from lodash, and convert it to a set at the end.
So:
```
const puIdsToIncludeFromIdsLessIdsToExcludeFromGeo = difference(puIdsToIncludeFromIds, puIdsToExcludeFromGeo);
const puIdsToExcludeFromIdsLessIdsToIncludeFromGeo = difference(puIdsToExcludeFromIds, puIdsToIncludeFromGeo);
```
and
```
const doInclusionAndExclusionIntersects = intersection(uniquePuIdsToInclude, uniquePuIdsToExclude).length > 0
```

Reply:
👍🏼 more or less done in aed5b088add61ffa63805f29e8dd6cd2e8d89084